editor/editor_inspector.cpp

```diff
@@ -905,47 +905,17 @@ void EditorProperty::_update_pin_flags() {
 	}
 }
 
-static Control *make_help_bit(const String &p_item_type, const String &p_text, const String &p_warning, const Color &p_warn_color) {
-	// `p_text` is expected to be something like this:
-	// `item_name|Item description.`.
-	// Note that the description can be empty or contain `|`.
-	PackedStringArray slices = p_text.split("|", true, 1);
-	if (slices.size() < 2) {
-		return nullptr; // Use default tooltip instead.
-	}
-
-	String item_name = slices[0].strip_edges();
-	String item_descr = slices[1].strip_edges();
-
-	String text;
-	if (!p_item_type.is_empty()) {
-		text = p_item_type + " ";
-	}
-	text += "[u][b]" + item_name + "[/b][/u]\n";
-	if (item_descr.is_empty()) {
-		text += "[i]" + TTR("No description.") + "[/i]";
-	} else {
-		text += item_descr;
-	}
-	if (!p_warning.is_empty()) {
-		text += "\n[b][color=" + p_warn_color.to_html(false) + "]" + p_warning + "[/color][/b]";
-	}
-
-	EditorHelpBit *help_bit = memnew(EditorHelpBit);
-	help_bit->get_rich_text()->set_custom_minimum_size(Size2(360 * EDSCALE, 1));
-	help_bit->set_text(text);
-
-	return help_bit;
-}
-
 Control *EditorProperty::make_custom_tooltip(const String &p_text) const {
-	String warn;
-	Color warn_color;
+	EditorHelpTooltip *tooltip = memnew(EditorHelpTooltip(p_text));
+
 	if (object->has_method("_get_property_warning")) {
-		warn = object->call("_get_property_warning", property);
-		warn_color = get_theme_color(SNAME("warning_color"));
+		String warn = object->call("_get_property_warning", property);
+		if (!warn.is_empty()) {
+			tooltip->set_text(tooltip->get_rich_text()->get_text() + "\n[b][color=" + get_theme_color(SNAME("warning_color")).to_html(false) + "]" + warn + "[/color][/b]");
```

Comment:
I think it would probably be better to add a dedicated method and store the warning itself separately, when apply it when you format the text.

To avoid passing the color you can get it through `EditorNode::get_editor_theme()` from `EditorHelpTooltip`.

Reply:
You mean making adding the warning a built-in feature of `EditorHelpTooltip`? I really don't see the need for something that is done only once in a single class and that's barely 3 lines long.

Reply:
Yes, that what I mean. It's not to save on lines, but to make the usage clearer by hiding implementation detail. All this boilerplate hides the fact that all we want to do is set some warning text at the end of the original message.

Reply:
I will have to disagree on this one, this is adding extra complexity to `EditorHelpTooltip` for a literal single use case.

Reply:
Eh, I'm not going to die on this hill (I can always rework this later myself anyway), but use cases for `EditorHelpTooltip` are already a handful, so it's not as outrageous as your response might suggest 🙃

Reply:
Didn't mean to suggest it was outrageous, just that it wasn't needed. If more cases of warnings needing to be added to the tooltip appear, I would gladly add the feature myself! :stuck_out_tongue: